dlt/extract/resource.py

```diff
@@ -310,10 +310,16 @@ def _gen_wrap(gen: TPipeStep) -> TPipeStep:
             count = 0
             if inspect.isfunction(gen):
                 gen = gen()
+
+            # wrap async gen already here
+            if inspect.isasyncgen(gen):
+                gen = wrap_async_generator(gen)
+
             try:
                 for i in gen:  # type: ignore # TODO: help me fix this later
                     yield i
-                    count += 1
+                    if i is not None:
```

Comment:
this line is needed for the async generator to work properly, it changes the behavior of the limit, but probably that is ok, i am not sure.